src/biomechanic/interpolate_points.jl

```diff
@@ -9,57 +9,54 @@ Interpolate points along a curve.
 
 # Returns
 - A NamedTuple with fields:
-  - `vec_x`: Vector of interpolated x coordinates.
-  - `vec_y`: Vector of interpolated y coordinates.
-  - `vec_z`: Vector of interpolated z coordinates.
+  - `points`: Meshes.Points of the interpolated points.
   - `i_discret_pts_exp`: Vector of indices of experimental points in the discretized curve.
   - `vec_dist_p2p1`: Vector of distances between consecutive interpolated points.
   - `vec_angle_xy`: Vector of angles between segments and the XY plane.
   - `vec_angle_xz`: Vector of angles between segments and the XZ plane.
 """
-function interp_points(x, y, z, pas)
+function interp_points(points, pas)
     # Distance and angles of each segment P2P1
-    vdist_p2p1, vangle_xy, vangle_xz = xyz_to_dist_and_angles(x, y, z)
+    vdist_p2p1, vangle_xy, vangle_xz = xyz_to_dist_and_angles(points)
 
     dist_lineique = cumsum(vdist_p2p1)
     dist_totale = last(dist_lineique)
 
     # The distances of the segments cannot be zero
     # The origin point (0,0,0) cannot be in mat_points
-    if any(vdist_p2p1 .== 0)
+    if any(vdist_p2p1 .== zero(vdist_p2p1[1]))
         error("Found distances between segments equal to 0.")
     end
 
     # Construction of interpolated points
     nlin = round(Int, dist_totale / pas + 1)
     pas = dist_totale / (nlin - 1)
 
-    npoints_exp = length(x)
+    npoints_exp = length(points)
 
-    vec_dist = [0; fill(pas, (nlin - 1))]
+    vec_dist = [zero(eltype(pas)); fill(pas, (nlin - 1))]
     dist_interp = cumsum(vec_dist)
 
     # Avoid rounding errors
-    dist_lineique[npoints_exp] = dist_lineique[npoints_exp] + 1
+    dist_lineique[npoints_exp] = dist_lineique[npoints_exp] + (1.0 * unit(eltype(dist_lineique)))
 
     mat_xyz = zeros(3, 0)
 
-    for iter in 1:npoints_exp
-        if iter == 1
-            ind_points = (dist_interp .<= dist_lineique[iter])
+    for i in 1:npoints_exp
+        if i == 1
+            ind_points = (dist_interp .<= dist_lineique[i])
         else
-            ind_points = (dist_interp .> dist_lineique[iter-1]) .& (dist_interp .<= dist_lineique[iter])
+            ind_points = (dist_interp .> dist_lineique[i-1]) .& (dist_interp .<= dist_lineique[i])
         end
 
         if !any(ind_points)
             error("No point found")
         end
 
         dist_points = vec_dist[ind_points]
-        op = [dist_points'; zeros(2, length(dist_points))]
-        vec_rot = rota_yz(op, vangle_xy[iter], vangle_xz[iter])
+        vec_rot = RotYZ(-vangle_xy[i], -vangle_xz[i]) * [dist_points'; zeros(eltype(dist_points), 2, length(dist_points))]
```

Comment:
Also, why did you change the values of the angles ?
```suggestion
        vec_rot = RotYZ(vangle_xy[i], vangle_xz[i]) * [dist_points'; zeros(eltype(dist_points), 2, length(dist_points))]
```

Reply:
I changed the angles because we changed them in `rota_inverse_yz`, so it was clearly an error here (but not for the changes based on `rota_inverse_yz`). Also, your comment made me check the order of rotations in `rota_yz`, and we should use `RotZY`, not `RotYZ` as we should start with rotating around Y and then Z.